compiler/rustc_data_structures/src/sync/parallel.rs

```diff
@@ -167,6 +176,26 @@ mod enabled {
         });
     }
 
+    pub fn try_par_for_each_in<
```

Comment:
This does look at bit expensive. It's probably a good idea to measure the performance impact of it. It might be a better idea to write the error to a mutex / atomic instead.

Reply:
Oof. I'll build some benchmarks. Seems bad if rayon's fold is more expensive than doing this manually.